### PR TITLE
RUN-2612: Functional Tests for Job Listing API

### PR DESCRIPTION
**Pull request by mrdubr**
Additional functional tests to test the [project's jobs](https://docs.rundeck.com/docs/api/#listing-jobs)  API.

### CODE REVIEW

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobsSpec.groovy [hunk 1]
```diff
@@ -6,28 +6,71 @@ import org.rundeck.util.annotations.APITest
...
+ */
 @APITest
 class JobsSpec extends BaseContainer {
 
+    static final def newProject = UUID.randomUUID().toString()
+    static List<String> createdJobIds
+
     def setupSpec() {
         startEnvironment()
         setupProject()
```
Comment (smartinellibenedetti) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobsSpec.groovy:20]:
```suggestion
```

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobsSpec.groovy [hunk 2]
```diff
@@ -6,28 +6,71 @@ import org.rundeck.util.annotations.APITest
...
-    def "Listing RunDeck Jobs for project test"() {
+    def cleanupSpec() {
+        deleteProject(newProject)
+    }
+
+    def "Listing jobs for a project that has no jobs"() {
         when:
-            def response = doGet("/project/${PROJECT_NAME}/jobs")
+        def aProject = UUID.randomUUID().toString()
+        setupProject(aProject)
```
Comment (smartinellibenedetti) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobsSpec.groovy:34]: It shouldn't be necessary to call setupProject here again

**Review by smartinellibenedetti — APPROVED** [review]
Looks good to me, the only change necessary is to remove the extra setupProject calls. I would suggest adding a column to the table with a description of what is being tested to make it easier to identify the cases in the future, but that's totally optional.

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobsSpec.groovy (hunk 2 above)
Comment (mrdubr) [reply functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobsSpec.groovy:34]: That's another project being set up in addition to the one in setupSpec()

**Comment by mrdubr** [comment]
> Looks good to me, the only change necessary is to remove the extra setupProject calls. I would suggest adding a column to the table with a description of what is being tested to make it easier to identify the cases in the future, but that's totally optional.

That's an excellent suggestion. I will keep this in mind for the next time. Thanks for the suggestion.

**Pull request merged**